src/data/providers.rs

```diff
@@ -16,54 +15,48 @@ pub struct Provider {
     pub provider_type: ProviderType,
     pub connection_status: ProviderConnectionStatus,
     pub enabled: bool,
-    pub models: Vec<RemoteModelId>,
+    pub models: Vec<BotId>,
     /// Whether the provider was added by the user or not
     pub was_customly_added: bool,
 }
 
 /// Creates a client for the provider based on the provider type
 pub fn create_client_for_provider(provider: &Provider) -> Box<dyn ProviderClient> {
     match &provider.provider_type {
-        ProviderType::OpenAI => Box::new(OpenAIClient::new(provider.url.clone(), provider.api_key.clone())),
+        ProviderType::OpenAI | ProviderType::MolyServer => Box::new(OpenAIClient::new(provider.url.clone(), provider.api_key.clone())),
         ProviderType::MoFa => Box::new(MofaClient::new(provider.url.clone())),
         ProviderType::DeepInquire => Box::new(DeepInquireClient::new(provider.url.clone(), provider.api_key.clone())),
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Hash, Eq)]
-pub struct RemoteModelId(pub String);
-
-impl RemoteModelId {
-    pub fn from_model_and_server(agent_name: &str, server_address: &str) -> Self {
-        RemoteModelId(format!("{}-{}", agent_name, server_address))
-    }
-}
-
-#[derive(Debug, Default, Clone, Hash, Eq, PartialEq, Serialize, Deserialize)]
-pub struct RemoteServerId(pub String);
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct RemoteModel {
-    pub id: RemoteModelId,
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ProviderBot {
+    pub id: BotId,
     pub name: String,
     pub description: String,
     pub provider_url: String,
     pub enabled: bool,
 }
```

Comment:
If we want to add `enabled` flag to MolyKit's `Bot` (and actually use it) then we can get rid of this struct on the App side (at least for now we don't need any other extras)